doc/coq.rst

```diff
@@ -104,9 +110,11 @@ The semantics of the fields are:
   theory, which means the ``Coq`` prefix won't be bound, and ``Coq.Init.Prelude``
   won't be imported by default.
 
-- The path to the installed locations of the ``<ocaml_plugins>`` is passed to
-  ``coqdep`` and ``coqc`` using Coq's ``-I`` flag. This allows a Coq theory to
-  depend on OCaml plugins.
+- If ``plugins`` is present, Dune will pass the corresponding flags to
+  Coq so that ``coqdep`` and ``coqc`` can find the corresponding OCaml
+  libraries declared in ``<ocaml_plugins>``. This allows a Coq theory
+  to depend on OCaml plugins. Starting with ``(lang coq 0.6)``,
```

Comment:
I thought we checked for private plugins before 0.6? I may be confused however. I seem to remember introducing the limitation in the workspace composition PR.

Reply:
I think that was only checked if the plugins live in a different scope.

With the legacy loading method it is fine to have private libs as plugins, only now that we depend (unconditionally) on the META file this is a hard limitation.

Reply:
Private plugins don't make any sense currently in Coq (only for private theories). If you have a public theory then you can install .vo files, however they will rely on the plugin to be installed somewhere to work hence it doesn't make sense to support this.

Reply:
Not sure I understand what you mean. Indeed it is logical that private plugins do only make sense for private theories, but that's a valid use case we could support.

Reply:
Right I was talking about before. Having private libs in private theories was the only use case that made sense. Private libs in public theories doesn't make any sense. Now with the new loading method however, it means that even private theories can only import public libs which is not currently restricted by Dune IIRC.

To allow for the user case you mention, we would need to yet again change how Coq loads plugins.

Reply:
We don't need to change how Coq loads plugins, it is enough for Dune to setup a virtual findlib env where the private libraries are exposed. But that's quite some work tho.